src/index.d.ts

```diff
@@ -59,6 +59,14 @@ declare class DataLoader<K, V, C = K> {
    * change is made. Returns itself for method chaining.
    */
   prime(key: K, value: V | PromiseLike<V> | Error): this;
+
+
+  /**
+   * The name given to this `DataLoader` instance. Useful for APM tools.
+   *
+   * Is `null` if not set in the constructor.
+   */
+  name: string | null;
```

Comment:
this can also be `undefined` given in config it is `name?: string | null` if you don't give a `name` then that is `undefined`

Reply:
Actually, the constructor then defaults to null - https://github.com/graphql/dataloader/blob/main/src/index.js#L478-L484

The config also specifically mentions this defaulting - https://github.com/graphql/dataloader/blob/main/src/index.d.ts#L124